examples/cluster-with-resources/README.md

```diff
@@ -0,0 +1,63 @@
+# Provision a cluster on AKS and install Kubernetes manifests
+
+This example demonstrates how to use the provider together with a cluster provisioned on Azure's AKS.
+
+**It's important to be aware that this example will not work as expected when applied with Terraform as a single operation.**
+
+This is because the provider requires access to the Kubernetes API during the planning phase. Initially, before any resources have been created, there is no available API endpoint and the provider will throw an error.
+
+This example demonstrates how to group resources related to the cluster in their own module and the Kubernetes manifests in another separate module, to facilitate building in distinct stages.
+
+It is still possible to build this configuration and obtain a single state file. For this, two Terraform operations are necessary.
+
+Before anything, make sure your environment is configured with valid Azure credentials.
```

Comment:
Sure, the more detail the better. 
My initial intention was to defer the details of the Azure setup to the Azure provider docs in case they change in time and what's mentioned here grows stale. But there a pretty slim chance Azure would change their credential schema anytime soon.